domain/application/state/state.go

```diff
@@ -1270,6 +1271,62 @@ ORDER BY array_index ASC;
 	return result, nil
 }
 
+func (st *State) checkApplicationNameExists(ctx context.Context, tx *sqlair.TX, name string) error {
+	app := applicationDetails{Name: name}
+	existsQueryStmt, err := st.Prepare(`
+SELECT &applicationDetails.uuid
+FROM application
+WHERE name = $applicationDetails.name
+`, app)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	if err := tx.Query(ctx, existsQueryStmt, app).Get(&app); err != nil {
+		if errors.Is(err, sqlair.ErrNoRows) {
+			return nil
+		}
+		return fmt.Errorf("checking if application %q exists: %w", name, err)
+	}
+	return applicationerrors.ApplicationAlreadyExists
```

Comment:
This method name is `checkApplicationNameExists`, but really it checks that application name doesn't exist?

Maybe `checkApplicationNameAvailable`? Or `Unused`?